lightkube/core/internal_models.py

```diff
@@ -1,7 +1,7 @@
 import sys
 
 try:
-    from ..models import meta_v1, autoscaling_v1
+    from ..models import meta_v1, autoscaling_v1, apiextensions_v1
```

Comment:
k8s 1.15 doesn't have `apiextensions_v1` but only [apiextensions_v1beta1](https://gtsystem.github.io/lightkube-models/1.15/resources/apiextensions_v1beta1/#customresourcedefinition). Some refactoring may be needed here in order to fallback to `apiextensions_v1beta1`.

Something like (not tested):

```python
...

try:
    from ..models import meta_v1, autoscaling_v1   # check if models are available at all, import what is always present.

    try:
      from ..models import apiextensions_v1 as apiextensions   # ensure we can always import apiextensions with the same name
    except:
      from ..models import apiextensions_v1beta1 as apiextensions
except:
    if sys.modules["__main__"].__package__ != 'mkdocs':   # we ignore this import error during documentation generation
        raise
    from unittest import mock

    ... mocks go here...
```

Reply:
I think I've got this handled now as you were suggesting, where `generic_resource.py` just uses whichever `apiextensions` it can from `internal_models.py`